### PR TITLE
Bump sigs.k8s.io/node-feature-discovery to v0.16.6

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
This change reverts #1007 and bumps the NFD client packages to v0.16.6.

### CODE REVIEW

**Comment by ArangoGutierrez** [comment]
As discussed we can now revert #1007

**Comment by ArangoGutierrez** [comment]
I also created https://github.com/NVIDIA/k8s-device-plugin/pull/1025 - that is the step required to revert the #1007 changes

**Comment by elezar** [comment]
> I also created #1025 - that is the step required to revert the #1007 changes

I don't agree. Dependabot doesn't update helm subcharts and only updates go imports in this context. Note that since the packages have been moved in NFD between v0.15.x and v0.16.x we would expect to have to update these manually.

File: go.mod [hunk 1]
```diff
@@ -24,7 +24,8 @@ require (
 	k8s.io/klog/v2 v2.130.1
 	k8s.io/kubelet v0.31.1
 	k8s.io/mount-utils v0.31.1
-	sigs.k8s.io/node-feature-discovery v0.15.4
+	sigs.k8s.io/node-feature-discovery v0.16.6
+	sigs.k8s.io/node-feature-discovery/api/nfd v0.0.0-20241030145326-d13729bfa450
```
[the remark this answers is not in the document]
Comment (ArangoGutierrez) [reply go.mod]: It seems that we need to cut a tag for the submodule

[the remark this answers is not in the document]
Comment (elezar) [reply go.mod]: Yes: see, for example https://github.com/cncf-tags/container-device-interface/tags

**Pull request merged**